applications/GeoMechanicsApplication/tests/cpp_tests/test_apply_c_phi_reduction_process.cpp

```diff
@@ -0,0 +1,105 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Anne van de Graaf
+//
+#include "containers/model.h"
+#include "custom_processes/apply_c_phi_reduction_process.hpp"
+#include "custom_constitutive/linear_elastic_plane_strain_2D_law.h"
+#include "geometries/quadrilateral_2d_4.h"
+#include "processes/structured_mesh_generator_process.h"
+#include "stub_linear_elastic_law.h"
+#include "testing/testing.h"
+#include <boost/numeric/ublas/assignment.hpp>
+
+#include <boost/algorithm/cxx11/all_of.hpp>
+#include <boost/algorithm/cxx11/none_of.hpp>
+
+using namespace Kratos;
+
+namespace
+{
+ModelPart& PrepareCPhiTestModelPart(Model& rModel)
+{
+    auto& result = rModel.CreateModelPart("dummy");
+
+    // Set up the test model part mesh
+    auto                   p_point_1 = Kratos::make_intrusive<Node>(1, 0.0, 0.0, 0.0);
+    auto                   p_point_2 = Kratos::make_intrusive<Node>(2, 0.0, 1.0, 0.0);
+    auto                   p_point_3 = Kratos::make_intrusive<Node>(3, 1.0, 1.0, 0.0);
+    auto                   p_point_4 = Kratos::make_intrusive<Node>(4, 1.0, 0.0, 0.0);
+    Quadrilateral2D4<Node> domain_geometry(p_point_1, p_point_2, p_point_3, p_point_4);
+
+    Parameters mesher_parameters(R"({
+        "number_of_divisions": 2,
+        "element_name": "Element2D3N",
+        "condition_name": "LineCondition",
+        "create_skin_sub_model_part": true
+    })");
+
+    StructuredMeshGeneratorProcess(domain_geometry, result, mesher_parameters).Execute();
+
+    auto  p_dummy_law             = std::make_shared<Testing::StubLinearElasticLaw>();
+    auto& r_model_part_properties = result.GetProperties(0);
+    r_model_part_properties.SetValue(CONSTITUTIVE_LAW, p_dummy_law);
+    Vector umat_parameters(6);
+    umat_parameters <<= 10000000, 0.2, 10.0, 25.0, 25.0, 1000;
+    r_model_part_properties.SetValue(UMAT_PARAMETERS, umat_parameters);
+    r_model_part_properties.SetValue(INDEX_OF_UMAT_C_PARAMETER, 3);
+    r_model_part_properties.SetValue(INDEX_OF_UMAT_PHI_PARAMETER, 4);
+    r_model_part_properties.SetValue(NUMBER_OF_UMAT_PARAMETERS, 6);
+
+    return result;
+}
+
+void CheckReducedCPhi(ModelPart& rModelPart, double COrig, double PhiOrig, double ReductionFactor)
+{
+    block_for_each(rModelPart.Elements(), [COrig, PhiOrig, ReductionFactor](Element& rElement) {
```

Comment:
Since this is a unit test, I don't expected we gain any performance by using the parallel block_for_each. Here, I would just use a range based for-loop
```suggestion
    for(auto& rElement : rModelPart.Elements(){
```

Reply:
changed.